File: solutions/121.Best_Time_to_Buy_and_Sell_Stock/AC_simulatoin_n.py
#!/usr/bin/python
# -*- coding: utf-8 -*-
# Author:      illuz <iilluzen[at]gmail.com>
# File:        AC_simulatoin_n.py
# Create Date: 2015-03-12 10:51:45
# Usage:       AC_simulatoin_n.py 
# Descripton:  


class Solution:
    # @param prices, a list of integer
    # @return an integer
    def maxProfit(self, prices):
        if not prices:
            return 0
        buy_price, profit = prices[0], 0
        for price in prices:
            buy_price = min(buy_price, price)
            profit = max(profit, price - buy_price)
        return profit

# debug
s = Solution()
print s.maxProfit([])

File: solutions/121.Best_Time_to_Buy_and_Sell_Stock/AC_simulatoin_n.cpp
/*
*  Author:      illuz <iilluzen[at]gmail.com>
*  File:        AC_simulatoin_n.cpp
*  Create Date: 2015-03-12 10:46:21
*  Descripton:  profit = prices[sell] - prices[buy]
*/

#include <bits/stdc++.h>

using namespace std;
const int N = 0;

class Solution {
public:
    int maxProfit(vector<int> &prices) {
        int buy = 0;
        int profit = 0;
        for (int i = 0; i < prices.size(); ++i) {
            if (prices[buy] > prices[i])
                buy = i;
            profit = max(profit, prices[i] - prices[buy]);
        }
    }
};

int main() {

    return 0;
}

